tests/php/ORM/HierarchyTest.php

```diff
@@ -55,6 +55,31 @@ public function testPreventLoop()
     }
 
     /**
+     * Test Hierarchy doesn't delete all records when non in the database
+     */
+    public function testDeleteNonExistentObject()
+    {
+        $numObjs = HierarchyTest\TestObject::get()->count();
+        $this->assertNotEmpty($numObjs);
+
+        $obj = new HierarchyTest\TestObject();
+        $this->assertFalse($obj->exists());
+        $fail = true;
+        try {
+            $obj->delete();
```

Comment:
Please use expectException() / expectExceptionMessage() instead of custom error catching.